src/reason_parser.mly

```diff
@@ -3459,7 +3457,7 @@ many_type_declarations:
       let loc = mklocation $symbolstartpos $endpos in
       [Type.mk ident
        ~params:params ~cstrs:constraints
-       ~kind ~priv ?manifest ~attrs:(add_nonrec $2 $4 2) ~loc]
+       ~kind ~priv ?manifest ~attrs:(add_nonrec $2 $4 $startpos($2) $endpos($2)) ~loc]
```

Comment:
In the grammar rule above, we can say:
```
| TYPE as_loc(nonrec_flag) type_declaration_details post_item_attributes
```
This turns nonrec_flag into a "loc" object, and we can then call it with
```
~kind ~priv ?manifest ~attrs:(add_nonrec $2 $4) ~loc]
```